### PR TITLE
[Bug](udf) fix java-udaf process string type error and add some tests

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes

- add is_return_null in create java-udf function
- fix udaf process string and date type error
- add some tests about udaf

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
5. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
6. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
7. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
8. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.99 seconds
 load time: 509 seconds
 storage size: 17179320470 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221109062951_clickbench_pr_42526.html

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateFunctionStmt.java [hunk 1]
```diff
@@ -221,6 +228,19 @@ private void analyzeCommon(Analyzer analyzer) throws AnalysisException {
                 throw new AnalysisException("library's checksum is not equal with input, checksum=" + checksum);
             }
         }
+        if (binaryType == TFunctionBinaryType.JAVA_UDF) {
+            String returnNullModeStr = properties.get(IS_RETURN_NULL);
+            if (returnNullModeStr == null) {
```
Comment (Gabriel39) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateFunctionStmt.java:233]: Make it optional and ALWAYS_NULLABLE by default

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateFunctionStmt.java [hunk 2]
```diff
@@ -96,6 +97,8 @@ public class CreateFunctionStmt extends DdlStmt {
     public static final String MERGE_METHOD_NAME = "merge";
     public static final String GETVALUE_METHOD_NAME = "getValue";
     public static final String STATE_CLASS_NAME = "State";
+    // add for java udf check return type nullable mode, always_nullable or always_not_nullable
+    public static final String IS_RETURN_NULL = "is_return_null";
```
Comment (Gabriel39) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateFunctionStmt.java:101]:
```suggestion
    public static final String IS_RETURN_NULL = "always_nullable";
```

File: docs/zh-CN/docs/ecosystem/udf/java-user-defined-function.md [hunk 3]
```diff
@@ -74,16 +74,20 @@ PROPERTIES (["key"="value"][,...])
 1. PROPERTIES中`symbol`表示的是包含UDF类的类名，这个参数是必须设定的。
 2. PROPERTIES中`file`表示的包含用户UDF的jar包，这个参数是必须设定的。
 3. PROPERTIES中`type`表示的 UDF 调用类型，默认为 Native，使用 Java UDF时传 JAVA_UDF。
-4. name: 一个function是要归属于某个DB的，name的形式为`dbName`.`funcName`。当`dbName`没有明确指定的时候，就是使用当前session所在的db作为`dbName`。
+4. PROPERTIES中`is_return_null`表示的 UDF 返回结果中是否有可能出现NULL值，可以设定为true或false。
```
Comment (Gabriel39) [inline docs/zh-CN/docs/ecosystem/udf/java-user-defined-function.md:77]:
```suggestion
4. PROPERTIES中`always_nullable`表示的 UDF 返回结果中是否有可能出现NULL值，可以设定为true或false。
```

**Review by Gabriel39 — DISMISSED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**